pennylane/pauli/pauli_arithmetic.py

```diff
@@ -379,8 +380,9 @@ def hamiltonian(self, wire_order=None):
                 raise ValueError("Can't get the Hamiltonian for an empty PauliSentence.")
             return Hamiltonian([1], [Identity(wires=wire_order)])
 
+        wire_order = wire_order or self.wires
         return sum(
-            coeff * pw.hamiltonian(wire_order=list(self.wires)) for pw, coeff in self.items()
+            coeff * pw.hamiltonian(wire_order=list(wire_order)) for pw, coeff in self.items()
```

Comment:
Should tests be added to check that the wire ordering works as expected now? I'd assume that the tests never reordered the wires if they were all passing before.

Reply:
Good call

Reply:
Added 👍🏼